terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf

```diff
@@ -261,6 +261,32 @@ data "aws_iam_policy_document" "airflow_dev_eks_assume_role_policy" {
 
 }
 
+##### Airflow Dev IRSA
+data "aws_iam_policy_document" "airflow_dev_monitoring_inline_role_policy" {
+  statement {
+    sid    = ""
+    effect = "Allow"
+    resources = [
+      "arn:aws:s3:::airflow-monitoring/airflow-scheduling-testing/*",
+      "arn:aws:s3:::airflow-monitoring/"
+    ]
+    actions = ["s3:GetObject", "s3:ListBucket", "s3:PutObject", "s3:DeleteObject"]
+  }
```

Comment:
Couple of things:
* Please split this policy definition into 2 statements. One which grants list bucket to the whole bucket and the other which grants the object-level perms to the folder
* Have a look in the console of the `airflow-role` policy attached to the `airflow_dev_monitoring` role which is what I believe you're looking to replicate. It's quite a lot bigger. Are you slimming it down intentionally?

Reply:
We didn't intend to cut down on the policy definitions, we have updated to bring in-line with existing `airflow-role `policy.